src/runtime_src/core/tools/xbutil2/OO_AieRegRead.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+* Copyright (C) 2021 Xilinx, Inc
+*
+* Licensed under the Apache License, Version 2.0 (the "License"). You may
+* not use this file except in compliance with the License. A copy of the
+* License is located at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations
+* under the License.
+*/
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "OO_AieRegRead.h"
+#include "tools/common/XBUtilities.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+namespace XBU = XBUtilities;
+
+// 3rd Party Library - Include Files
+#include <boost/program_options.hpp>
+#include <boost/algorithm/string/join.hpp>
+#include <boost/format.hpp>
+namespace po = boost::program_options;
+namespace qr = xrt_core::query;
+
+// System - Include Files
+#include <iostream>
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+OO_AieRegRead::OO_AieRegRead( const std::string &_longName, bool _isHidden )
+    : OptionOptions(_longName, _isHidden, "Read from the given memory address" )
+    , m_device("")
+    , m_row(0)
+    , m_col(0)
+    , m_reg("")
+    , m_help(false)
+{
+  m_optionsDescription.add_options()
+    ("device,d", po::value<decltype(m_device)>(&m_device), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
+    ("row", po::value<decltype(m_row)>(&m_row)->required(), "Row of core tile")
+    ("col", po::value<decltype(m_col)>(&m_col)->required(), "Column of core tile")
+    ("reg", po::value<decltype(m_reg)>(&m_reg)->required(), "Register name to read from core tile")
+    ("help,h", po::bool_switch(&m_help), "Help to use this sub-command")
```

Comment:
Being that you had earlier created a great map of the register name and offsets, wouldn't it provide a great EoU to print out what registers are available.   Do to the large quantity of registers, this would be best to be done via an help extension option where the output would be sorted in multiple columns.

Reply:
giving list of 300 register is not a good option we need to find another way to give this info to user

Reply:
Hmm...Agree that it is a lot of information, but imagine the frustration if they were use to typing in:
`Stream_Switch_Master_Config_South0`
And decide to read in:
`Stream_Switch_Slave_South_0_Config`
And didn't realize that there is an "underscore" needed between "South" and "0" in the second example and not the first.

While I know it isn't great, printing out valid registers options would help out greatly.   To "reduce" the output to 100 lines, you could print them out in 3 columns.